src/main/java/org/logstash/dissect/fields/AbstractField.java

```diff
@@ -0,0 +1,69 @@
+package org.logstash.dissect.fields;
+
+import com.logstash.Event;
+import org.logstash.dissect.Delimiter;
+import org.logstash.dissect.ValueResolver;
+
+import java.util.Map;
+
+public abstract class AbstractField implements Field {
+    private final int ordinal;
+    protected final String name;
+
+    protected Delimiter join;
+    protected Delimiter next;
```

Comment:
Technically, the `proctected` access modifier is correct but it is hard to reason about modifications to a field. You have to check / know the code of all subclasses to know who is modifying what and that could get out of hand quickly. Can you encapsulate all the logic in the base class instead maybe?